app/controllers/api/v1/mixins/rbac_mixin.rb

```diff
@@ -52,7 +52,7 @@ def permission_check(verb, klass = controller_name.classify.constantize)
         def resource_accessible?(resource, verb)
           return true if admin?
 
-          approver? ? permitted?(APPROVER_PERMISSIONS, resource, verb) : permitted?(OWNER_PERMISSIONS, resource, verb)
+          approver? && permitted?(APPROVER_PERMISSIONS, resource, verb) || permitted?(OWNER_PERMISSIONS, resource, verb)
```

Comment:
@bzwei Would it be more performant to swap the order of these checks and do `permitted?(OWNER_PERMISSIONS, resource, verb)` first and `approver? && permitted?(APPROVER_PERMISSIONS, resource, verb)` only if needed?